client/onboarding-prototype/steps/test/loading.tsx

```diff
@@ -0,0 +1,88 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { render } from '@testing-library/react';
+import apiFetch from '@wordpress/api-fetch';
+/**
+ * Internal dependencies
+ */
+import Loading from '../loading';
+
+// Mock Api Fetch module and function
+jest.mock( '@wordpress/api-fetch', () => jest.fn() );
+const mockApiFetch = apiFetch as jest.MockedFunction< typeof apiFetch >;
+
+// Mock wcpaySettings
+declare const global: {
+	wcpaySettings: {
+		connectUrl: string;
+	};
+};
+
+// Mock data, setData from OnboardingContext
+let data = {};
+const setData = jest.fn();
+
+jest.mock( '../../context', () => ( {
+	useOnboardingContext: jest.fn( () => ( {
+		data,
+		setData,
+	} ) ),
+} ) );
+
+describe( 'Loading', () => {
+	const originalWindowLocation = window.location;
+
+	beforeEach( () => {
+		// Prevent window.location.href redirect
+		Object.defineProperty( window, 'location', {
+			configurable: true,
+			enumerable: true,
+			value: new URL( window.location.href ),
+		} );
+		global.wcpaySettings = {
+			connectUrl: 'http://wcpay-connect-url',
+		};
+	} );
+
+	afterEach( () => {
+		// Roll back window.location.href behavior after test
+		Object.defineProperty( window, 'location', {
+			configurable: true,
+			enumerable: true,
+			value: originalWindowLocation,
+		} );
+	} );
+
+	it( 'renders loading screen and sends request to server', async () => {
```

Comment:
Async/await is not required here, `expect` is not a promise. And we're only testing that the function is called, not waiting for a response 🙂

Reply:
@ismaeldcom good ctach! so here we have two options - wait for the response and check URL is formed correct and another one is just omit it. I think we should check that URL has progressive param - then test will be full
What do you think?

Reply:
Testing also that the URL contains the progressive param sounds better 🙂

Reply:
Done in https://github.com/Automattic/woocommerce-payments/pull/5779/commits/2cdb4cd8bd4069a4cfd850d6d693e9c4f73da3eb